node/primitives/src/lib.rs

```diff
@@ -51,6 +52,9 @@ pub use disputes::{
 	SignedDisputeStatement, UncheckedDisputeMessage, ValidDisputeVote,
 };
 
+const MERKLE_NODE_MAX_SIZE: usize = 347;
+const MERKLE_PROOF_MAX_DEPTH: usize = 3;
```

Comment:
Note that these constants would not work in practice. I've just set them to what I observed being crated in the tests. It is my belief that the depth is no more than 3 (in fact it's not more than 2) throughout all tests because we don't ever create a Merkle Tree with that many elements, i.e. intermediate nodes.

The correct values to use here are:

MERKLE_NODE_MAX_SIZE = 512;
MERKLE_PROOF_MAX_DEPTH = 8;

I just found it interesting that the node max size was uneven. I may be mistaken, but I'm quite certain the max proof node created is 347 bytes.